src/securemul.rs

```diff
@@ -76,7 +79,7 @@ impl<F: Field> SecureMul<F> {
         let DValue {
             d: left_d,
             index: left_index,
-        } = ctx.helper_ring.receive(HelperAddr::Left).await?;
+        } = ctx.helper_ring.receive(HelperAddr::Left, self.step).await?;
```

Comment:
Do we expect the records to always come in order for this to work or, we plan to handle ordering in the Ring communication itself? (referring to IndexMismatch in the code on line 92 below)

Reply:
My expectation is that there will be a total ordering of items that are sent out that the communication infrastructure only needs to preserve.  You will determine (by setting a record number) what the order should be.  The changes we're talking about here should remove the IndexMismatch error ultimately.